lib/src/facts/windows/uptime_resolver.cc

```diff
@@ -23,28 +24,10 @@ namespace facter { namespace facts { namespace windows {
     {
     }
 
-    static ptime get_ptime(string const& wmitime)
-    {
-        static boost::regex wmi_regex("^(\\d{8,})(\\d{2})(\\d{2})(\\d{2})\\.");
-        string iso_date;
-        int hour, min, sec;
-        if (!re_search(wmitime, wmi_regex, &iso_date, &hour, &min, &sec)) {
-            throw runtime_error(_("failed to parse {1} as a date/time", wmitime));
-        }
-
-        return ptime(from_undelimited_string(iso_date), time_duration(hour, min, sec));
-    }
-
     int64_t uptime_resolver::get_uptime()
     {
-        auto vals = _wmi->query(wmi::operatingsystem, {wmi::lastbootuptime, wmi::localdatetime});
-        if (vals.empty()) {
-            return -1;
-        }
-
-        ptime boottime = get_ptime(wmi::get(vals, wmi::lastbootuptime));
-        ptime now = get_ptime(wmi::get(vals, wmi::localdatetime));
-        return (now - boottime).total_seconds();
+        uint64_t tickCount = GetTickCount64();
```

Comment:
This wasn't available in Windows Server 2003. Yay for using it now. 👍

Reply:
2 ^ 64 = 1.844674407370955e19 converted to seconds by dividing by 1000 = 1.844674407370955e16
2 ^ 63 = 9.223372036854776e18

Therefore count in seconds does not exceed range of allowed values in `int64_t` (2 ^63) ... so the cast should work fine here.